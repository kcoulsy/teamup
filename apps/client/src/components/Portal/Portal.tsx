import React from 'react';
import ReactDOM from 'react-dom';

interface ModalProps {
  children: React.ReactNode;
}

/**
 * https://reactjs.org/docs/portals.html
 */
class Portal extends React.Component<ModalProps> {
  private modalRoot: HTMLElement;

  private el: HTMLDivElement;

  constructor(props: ModalProps) {
    super(props);
    this.modalRoot = document.getElementById('modal-root') || document.createElement('div');
    this.el = document.createElement('div');
  }

  componentDidMount() {
    this.modalRoot.appendChild(this.el);
  }

  componentWillUnmount() {
    this.modalRoot.removeChild(this.el);
  }

  render() {
    const { children } = this.props;
    return ReactDOM.createPortal(children, this.el);
  }
}

export default Portal;
