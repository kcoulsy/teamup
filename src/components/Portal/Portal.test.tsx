import React from 'react';
import { shallow, mount, ShallowWrapper, ReactWrapper } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Portal from './Portal';

// TODO - update to react testing library
describe('render Portal component', () => {
  // const childJSX = <p>Child JSX</p>;
  // let component: ReactWrapper;
  // const modalRoot = global.document.createElement('div');
  // modalRoot.setAttribute('id', 'modal-root');
  // const body = global.document.querySelector('body');
  // body?.appendChild(modalRoot);
  // afterEach(() => {
  //     if (component) {
  //         component.unmount();
  //     }
  // });
  //   it('should render correctly', () => {
  // component = mount(<Portal>{childJSX}</Portal>);
  // expect(mountToJson(component)).toMatchSnapshot();
  //   });
});
