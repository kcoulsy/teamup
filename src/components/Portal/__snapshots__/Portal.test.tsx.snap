// Vitest Snapshot v1

exports[`render Portal component should render correctly 1`] = `
<Portal>
  <Portal
    containerInfo={
      <div>
        <p>
          Child JSX
        </p>
      </div>
    }
  >
    <p>
      Child JSX
    </p>
  </Portal>
</Portal>
`;
